src/installation/guides/zfs.md

```diff
@@ -0,0 +1,227 @@
+xbps-install -S -R $REPO grub zfs`
+
+## GRUB Installation
+
+Confirm the ZFS modules are loaded in the new system using
+[lsmod(8)](https://man.voidlinux.org/lsmod.8):
+
+```
+(chroot)$ lsmod | grep zfs
+zfs                  4022272  1
+zunicode              335872  1 zfs
+zlua                  172032  1 zfs
+zavl                   16384  1 zfs
+icp                   294912  1 zfs
+zcommon                86016  2 zfs,icp
+znvpair                69632  2 zfs,zcommon
+spl                   102400  5 zfs,icp,znvpair,zcommon,zavl
+```
+
+Setup the ZFS cachefile, to ensure the new machine recognizes the previously
+created pool properly.
+
+`(chroot)# zpool set cachefile=/etc/zfs/zpool.cache zroot`
+
+Then notate the bootable system for GRUB's autoconfig:
+
+`(chroot)# zpool set bootfs=zroot/ROOT/void zroot`
+
+Finally ensure GRUB recognizes the ZFS module before being installed.
+
+Note: When using id's the ZPOOL_VDEV_NAME_PATH variable must be set to 1.
+
+```
+(chroot)# ZPOOL_VDEV_NAME_PATH=1 grub-probe /
+zfs
+```
+
+Install GRUB onto the drive specifying the path:
+
+```
+(chroot)# ZPOOL_VDEV_NAME_PATH=1 grub-install /dev/sda
+Installing for i386-pc platform.
+Installation finished. No error reported.
+```
+
+Create and modify `/etc/dracut.conf.d/zol.conf` to configure Dracut to include
+at least the following values:
+
+```
+hostonly="yes"
+nofsck="yes"
+add_dracutmodules+=" zfs "
+```
+
+Reconfigure Linux to update GRUB's and Dracut's configurations:
+
+Note: If the Linux version on the system is unknown use a query to find the
+installed version on the new system: `xbps-query -s linux | grep modules`
+
+```
+(chroot)# xbps-reconfigure -f linux5.4
+linux5.4: configuring ...
+Executing post-install kernel hook: 10-dkms ...
+Available DKMS module: zfs-0.8.3.
+Building DKMS module: zfs-0.8.3... done.
+Executing post-install kernel hook: 20-dracut ...
+Executing post-install kernel hook: 50-grub ...
+Generating grub configuration file ...
+Found linux image: /boot/vmlinuz-5.4.32_1
+Found initrd image: /boot/initramfs-5.4.32_1.img
+Found void on /dev/sda2
+done
+linux5.4: configured successfully.
+```
+
+## Sanity Checks
+
+Ensure that the Dracut configuration took and the zfs module is listed with
+[lsinitrd(1)](https://man.voidlinux.org/lsinitrd):
+
+Note: It may be necessary to specify the path to the img if the kernel version
+has changed, see [lsinitrd(1)](https://man.voidlinux.org/lsinitrd).
+
+`(chroot)# lsinitrd -m`
+
+As well check that the ZFS cache is recognized:
+
+```
+(chroot)# lsinitrd | grep zpool.cache
+-rw-r--r--   1 root     root         1376 Apr 17 01:30 etc/zfs/zpool.cache
+```
+
+## Cleanup
+
+Exit chroot:
+
+`(chroot)$ exit`
+
+And un-mount the chroot environment:
+
+```
+# umount /mnt/void/dev/pts
+# umount /mnt/void/{dev,proc,sys}
+# zpool export zroot
+```
+
+Restart the system using [reboot(8)](https://man.voidlinux.org/reboot.8) and log
+in:
+
+`# reboot`
```

Comment:
In this case the code block for the command can be left out. You can just specify which command they need and that's enough.